state/remote/s3.go

```diff
@@ -188,3 +195,73 @@ func (c *S3Client) Delete() error {
 
 	return err
 }
+
+func (c *S3Client) Lock(reason string) error {
+	if c.lockTable == "" {
+		return nil
+	}
+
+	stateName := fmt.Sprintf("%s/%s", c.bucketName, c.keyName)
+
+	putParams := &dynamodb.PutItemInput{
+		Item: map[string]*dynamodb.AttributeValue{
+			"LockID":  {S: aws.String(stateName)},
+			"Created": {S: aws.String(time.Now().UTC().Format(time.RFC3339))},
+			"Expires": {S: aws.String(time.Now().Add(time.Hour).UTC().Format(time.RFC3339))},
```

Comment:
Could the expiration period be configurable somewhere? Maybe not necessarily from user's perspective (CLI), but I'm thinking it could be passed in as `time.Duration` argument to `Lock()`? That might also make developer implementing new remote backends realise that the expiration should be respected/implemented properly where possible.